src/lib.rs

```diff
@@ -148,25 +148,34 @@ impl ImportLibraryGenerator {
 
         write(&defpath, def_file_content)?;
 
+        let impllib_ext = if self.env == "msvc" {
+            IMPLIB_EXT_MSVC
+        } else {
+            IMPLIB_EXT_GNU
+        };
+        let implib_file = match self.version {
+            Some((major, minor)) => {
+                if self.env == "msvc" {
+                    format!("python{}{}{}", major, minor, impllib_ext)
+                } else {
+                    // https://packages.msys2.org/base/mingw-w64-python
+                    format!("python{}.{}{}", major, minor, impllib_ext)
+                }
```

Comment:
We need something less surprising for the users here...

Reply:
I also wonder whether this link is strictly necessary: I have been building extensions using MinGW but targeting the libraries from the MSVC-based upstream CPython releases.

Reply:
Oh, I just noticed that you explained all of this in https://github.com/PyO3/python3-dll-a/pull/17#issuecomment-1122607225. Sorry for noise.

Reply:
> I have been building extensions using MinGW but targeting the libraries from the MSVC-based upstream CPython releases.

Same here actually 😅